src/partest/scala/tools/partest/nest/DirectCompiler.scala

```diff
@@ -75,7 +75,7 @@ class DirectCompiler(val runner: Runner) {
     val Xplugin = if (xplugs.isEmpty) Nil else List(xprefix +
       (xplugs map (_ stripPrefix xprefix) flatMap (_ split pathSeparator) map absolutize mkString pathSeparator)
     )
-    runner.suiteRunner.scalacExtraArgs ++ runner.suiteRunner.scalacOpts.split(' ') ++ others ++ Xplugin
+    runner.suiteRunner.scalacExtraArgs ++ /*runner.suiteRunner.scalacOpts.split(' ') ++*/ others ++ Xplugin
```

Comment:
this should be cleaned up before merge

Reply:
It was documentary; but I'm not sure of the right thing to do. 

For example, when building with optimizer turned up, should tests be compiled that way? I know that some of the check files (such as output of -Xshow-phases) differs for that reason. My suspicion is that all test flags should be explicit. But I didn't track down what ExtraArgs is for.

Reply:
Seems this is actually in use? https://github.com/scala/scala/blob/67580388f35f200a1eb0d8ce424556bf5758b596/build.sbt#L1008

Disable this instead? https://github.com/scala/scala/blob/67580388f35f200a1eb0d8ce424556bf5758b596/build.sbt#L864

I agree it's weird to change the test flags depending on the compilation flags. Given that the optimizer is on by default on CI and for releases, enabling it for partest (unconditionally) might be the right thing to do?

Reply:
The macro-annotations case is handled by the flags file. 

You're suggesting always optimise [sic] for tests? I hadn't given any thought yet as to why that might be useful. Are we talking inline from ** kind of optimizer?

Obviously, I've been working on the -opt flags.

Reply:
> why that might be useful

As an integration test for the optimizer, that's what I can think of.

> inline from ** kind of optimizer

Yes (that's how partest currently runs in CI, I assume?)

Reply:
OK, so I'll remove the now-extraneous macro flag in the build, and filter the flags in partest to accept opt opts.